app/models/lr_common_config.rb

```diff
@@ -12,14 +12,22 @@ def app_link(app_name, request)
         css_classes << 'lr-header--header-proposition--a__active'
       end
 
+      # Sub replaces just the first instance of the app name in the relative_url_root
       path = relative_url_root.sub(%r{[^/]*\Z}, app_name)
+      # Add the lang param if it exists
       path += "?lang=#{request.params[:lang]}" if request.params.key?(:lang)
 
       link_to(I18n.t("common.app.#{app_name}"), path, class: css_classes.join(' '))
     end
 
+    # Returns the relative url root for the app if running in a subdirectory
+    # of the web server, or the root if it is not
+    # If the app is running in production, and the relative_url_root excludes
+    # /app, then the app name is appended to the relative_url_root of `/app`
     def relative_url_root
-      Rails.application.config.relative_url_root || '/'
```

Comment:
Why is this different in dev & production modes?
I might be running the behind a proxy in dev mode.

Reply:
@andrew-pickin-epi - as discussed, this now resolves the gem to leave configurations of the root path to be controlled in the individual applications and therefore the common styles gem should be "dumb" and fail if the required application is "mis-configured"